database/migrations/platform/2023_10_09_000000_update_player_tables_index.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        // add the inverse indexes for user_id so lookups for untracked and deleted are more likely using indexes
+
+        Schema::table('player_achievements', function (Blueprint $table) {
+            $sm = Schema::getConnection()->getDoctrineSchemaManager();
+            $indexesFound = $sm->listTableIndexes('player_achievements');
+
+            if (!array_key_exists('player_achievements_achievement_id_user_id_unlocked_hardcore_at', $indexesFound)) {
+                $table->index(
+                    ['achievement_id', 'user_id', 'unlocked_hardcore_at'],
+                    'player_achievements_achievement_id_user_id_unlocked_hardcore_at'
+                );
+            }
+        });
+
+        Schema::table('player_games', function (Blueprint $table) {
+            $sm = Schema::getConnection()->getDoctrineSchemaManager();
+            $indexesFound = $sm->listTableIndexes('player_games');
+
+            if (!array_key_exists('player_games_game_id_user_id_index', $indexesFound)) {
+                $table->index(['game_id', 'user_id']);
+            }
```

Comment:
isn't this handled by the `player_games_user_id_game_id_unique` unique key?

Reply:
That's what I thought! After some tinkering and `EXPLAIN` statements (added them to the PR description above) I found that the query optimizer will only consider the indexes to be used when they exist in this particular order.
Thinking about for a while I guess it makes sense but would've expected it to use the user_id_game_id unique index, too.